packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -56,10 +62,53 @@ class UnthemedGraphNG extends React.Component<GraphNGProps, GraphNGState> {
     this.state = {
       alignedDataFrame: alignedData,
       data: preparePlotData(alignedData),
-      config: preparePlotConfigBuilder(alignedData, props.theme, this.getTimeRange, this.getTimeZone),
     };
   }
 
+  componentDidMount() {
+    const { alignedDataFrame } = this.state;
+
+    this.panelContext = this.context as PanelContext;
+    const { eventBus } = this.panelContext;
+
+    const config = preparePlotConfigBuilder(
+      alignedDataFrame,
+      this.props.theme,
+      this.getTimeRange,
+      this.getTimeZone,
+      eventBus
+    );
+    this.setState({ config });
+
+    // Something like this will be required to get external events *into* uPlot...
+    this.subscriptions.push(
+      eventBus
+        .getStream(DataHoverEvent)
+        .pipe(
+          throttleTime(100),
+          filter((e) => e.origin !== eventBus)
+        )
+        .subscribe({
+          next: (evt) => {
+            const scales = this.state.config?.scaleKeys;
+            if (scales) {
+              const x = evt.payload.point[scales[0]];
+              const y = evt.payload.point[scales[1]];
+
+              // const pX = (x==null) ? -10 : client.valToPos(x, scales[0]);
+              console.log('DataHoverEvent //', x, y);
+            }
+          },
+        })
+    );
```

Comment:
@ryantxu refactored this to getStream, wanted to add some throttling and filter. I like the filter better than if(evt.origin === eventBus).

Anyways, either `filter` or `if` will be repeated across different panels in the future. Was thinking - maybe instead ef exposing entire `eventBus` on the context, we could instead expose API similar to this:

```
subscribe(evt: BusEventType, handler): Unsubscribable;
subscribeOwn(evt: BusEventType, handler): Unsubscribable;
```

Or extend the EventBus interface with `subscribeOwn` method that would do the filtering so that we don't repeat it over and over.

Reply:
Ah, forget ^^ - we have `onlyLocal` in the constructor of the scoped bus.